default.nix

```diff
@@ -178,6 +178,15 @@ in rec {
         --set GHC_RTS "-M2G"
     '';
 
+  webCommon = pkgs.lib.cleanSourceWith { 
+    filter = pkgs.lib.cleanSourceFilter;
+    src = lib.cleanSourceWith {
+      filter = (path: type: !(pkgs.lib.elem (baseNameOf path)
+                            [".spago" ".spago2nix" "generated" "generated-docs" "output" "dist" "node_modules" ".psci_modules" ".vscode"]));
```

Comment:
I think it's worth trying `pkgs.nix-gitignore.gitignoreFilter`.

https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-nix-gitignore-usage

Reply:
this is still a problem, I'm going to look into it further in another PR

Reply:
opened SCP-1328